### PR TITLE
Refactor to avoid using fork to create initial file

### PR DESCRIPTION
**Pull request by qkoziol**
It's not necessary to fork another process to create a file with only one process of an MPI job.

### CODE REVIEW

File: testpar/t_cache_image.c [hunk 1]
```diff
@@ -3937,50 +3657,28 @@ main(int argc, char **argv)
...
-                HDfprintf(stdout, "setup process reports failure.\n");
+            if (pass) {
+                HDprintf("done.\n");
+                HDfflush(stdout);
             }
             else {
-
-                HDfprintf(stdout, "testfile construction complete -- proceeding with tests.\n");
+                HDprintf("failed.\n");
+                HDexit(1);
```
Comment (derobins) [inline testpar/t_cache_image.c]: Better to use EXIT_SUCCESS and EXIT_FAILURE

Comment (qkoziol) [reply testpar/t_cache_image.c]: > Better to use EXIT_SUCCESS and EXIT_FAILURE

True, will update.

**Review by derobins — APPROVED** [review]

**Comment by qkoziol** [comment]
@here - Can this get reviewed today, so I can submit more async PRs tonight?  (I don't want to create more PRs until I've cleared through these, to avoid intersecting git merges and make it clearer what I have left to merge)

**Comment by byrnHDF** [comment]
So if I understand correctly, the fork was only used to create test files. This change then creates them inline before using them.

**Comment by qkoziol** [comment]
> So if I understand correctly, the fork was only used to create test files. This change then creates them inline before using them.

Yes, forking was unnecessary.

**Review by lrknox — APPROVED** [review]

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**